### PR TITLE
oauth-apiserver:workload: wait for storage urls before installing the api server

### PR DESCRIPTION
**Pull request by p0lyn0mial**

### CODE REVIEW

**Comment by p0lyn0mial** [comment]
/hold

requires changes in 'library-go'

**Comment by p0lyn0mial** [comment]
https://github.com/openshift/library-go/pull/955

File: vendor/github.com/openshift/library-go/pkg/operator/configobserver/etcd/observe_etcd_endpoints.go [hunk 1]
```diff
@@ -23,7 +23,9 @@ const (
 
 var (
 	OldStorageConfigURLsPath = []string{"storageConfig", "urls"}
-	StorageConfigURLsPath    = []string{"apiServerArguments", "etcd-servers"}
+	StorageConfigURLsPath    = []string{"apiServerArguments", StorageConfigURLsKey}
+	
+	StorageConfigURLsKey     = "etcd-servers"
```
Comment (p0lyn0mial) [inline vendor/github.com/openshift/library-go/pkg/operator/configobserver/etcd/observe_etcd_endpoints.go:28]: moved to library-go - https://github.com/openshift/library-go/pull/963

File: pkg/operator/workload/sync_openshift_oauth_apiserver.go [hunk 2]
```diff
@@ -102,6 +107,12 @@ func (c *OAuthAPIServerWorkload) PreconditionFulfilled() (bool, error) {
 		klog.Info("Waiting for observed configuration to be available")
 		return false, errors.New("waiting for observed configuration to be available (haven't found APIServerArguments in spec.ObservedConfig.Raw)")
 	}
+
+	// specifying etcd servers list is mandatory, without it the pods will be crashlooping, so wait for it.
+	if storageServers, _ := operatorCfg.APIServerArguments[libgoetcd.StorageConfigURLsKey]; len(storageServers) == 0 {
+		klog.Infof("Waiting for observed configuration to have mandatory %q", libgoetcd.StorageConfigURLsKey)
+		return false, errors.New(fmt.Sprintf("waiting for observed configuration to have mandatory %q", libgoetcd.StorageConfigURLsKey))
```
Comment (stlaz) [inline pkg/operator/workload/sync_openshift_oauth_apiserver.go:114]: "waiting for observed configuration field `apiServerArguments.%s` to be set"

maybe?

File: pkg/operator/workload/sync_openshift_oauth_apiserver.go [hunk 3]
```diff
@@ -102,6 +107,12 @@ func (c *OAuthAPIServerWorkload) PreconditionFulfilled() (bool, error) {
 		klog.Info("Waiting for observed configuration to be available")
 		return false, errors.New("waiting for observed configuration to be available (haven't found APIServerArguments in spec.ObservedConfig.Raw)")
 	}
+
+	// specifying etcd servers list is mandatory, without it the pods will be crashlooping, so wait for it.
+	if storageServers, _ := operatorCfg.APIServerArguments[libgoetcd.StorageConfigURLsKey]; len(storageServers) == 0 {
```
Comment (stlaz) [inline pkg/operator/workload/sync_openshift_oauth_apiserver.go:112]: `storageServers := operatorCfg.APIServerArguments[libgoetcd.StorageConfigURLsKey; len(storageServers) == 0`
should work just the same, shouldn't it? (leaving the ignored variable out)

Comment (p0lyn0mial) [reply pkg/operator/workload/sync_openshift_oauth_apiserver.go:112]: yes, If the key doesn't exist we will get an empty slice :)

File: pkg/operator/workload/sync_openshift_oauth_apiserver.go (hunk 2 above)
Comment (p0lyn0mial) [reply pkg/operator/workload/sync_openshift_oauth_apiserver.go:114]: sure, np

Comment (p0lyn0mial) [reply pkg/operator/workload/sync_openshift_oauth_apiserver.go:114]: better be explicit

**Pull request merged**